admin-perspective/olm/install-forcenamespace/install-forcenamespace.md

```diff
@@ -0,0 +1,53 @@
+# Operator-defined Installation Namespace
+
+Operator bundles can define what namespace they should be installed to, so that namespace is recommended to the user. The user can still change the installation namespace if desired.
```

Comment:
> The user can still change the installation namespace if desired.

I don't think the user be able to change the installation namespace to an openshift-* if monitoring is enabled and it's a community operator.

Reply:
@LiliC unfortunately it's been determined that we can't know whether an operator is a community operator or not because customers can create their own catalogs of content (and in fact must do so for disconnected installs).  So there's no clear concept of a "community catalog" of content that we can explicitly distrust.

Reply:
> I don't think the user be able to change the installation namespace to an openshift-* if monitoring is enabled and it's a community operator.

@LiliC even to `openshift-operators` namespace ?

Reply:
> @LiliC unfortunately it's been determined that we can't know whether an operator is a community operator or not because customers can create their own catalogs of content (and in fact must do so for disconnected installs). So there's no clear concept of a "community catalog" of content that we can explicitly distrust.

Can't we keep a list of red hat operators and verify against that list somehow? 🤔

Reply:
@LiliC at the end of the day, cluster-admins can install whatever they want, where-ever they want.  It's not worth doing exceptional things in the UI to prevent it when they can just drop to the CLI and do it.  The best we can do is warn them if they are about to do something bad, and explain the potential consequences (and not default them into it).

Reply:
Yes understood and agreed, but if we can detect which are not red hat operators, then we can at least in those cases give a warning and say we void support.

Reply:
>  if we can detect which are not red hat operators, then we can at least in those cases give a warning and say we void support.

i think the only way to do this is the whitelist you proposed and i'd hate to see us try to maintain that as more and more redhat products start producing operators.

Reply:
would it then make sense to just have a dummy checkbox which when user will check the label will be added to the created/updated namespace, together with a warning msg ?

Reply:
what do you mean by "dummy" checkbox?  otherwise you described is what i thought we were asking for.  (but the checkbox+warning should only show up when the operator asks for monitoring)

Reply:
> i think the only way to do this is the whitelist you proposed and i'd hate to see us try to maintain that as more and more redhat products start producing operators.

Yes agreed, as long as there is a warning to the users that is clear enough that installing anything outside of the redhat operators into an openshift- namespace is not a supported feature with the warning and checkbox, we are good to go. 👍

Reply:
> i think the only way to do this is the whitelist you proposed and i'd hate to see us try to maintain that as more and more redhat products start producing operators.

As mentioned during the architecture call, we have another suggestion: instead of maintaining a whitelist which apparently is a burden, from the perspective of the monitoring stack it is actually more important to have test coverage in an e2e origin integration test.

All operators installed in that e2e origin test that successfully pass telemetry and cluster installation can then be considered to be "supported". An operator which is not covered in this e2e test is considered "unsupported".

I believe this was also discussed out of band between @shawn-hurley and @LiliC and there exists a ticket for the OLM team to implement that e2e test in origin.

Just to reemphasize, we need to take care of two things mainly:

1. maintain/retain the stability guarantees of the cluster monitoring stack which is considered part of the core product, i.e. "Prometheus doesn't blow up if all of these OLM operators are suddenly sending metrics".
2. Have the ability to introspect telemetry payload to evaluate impact for supported monitorable OLM operators.

Reply:
> what do you mean by "dummy" checkbox? otherwise you described is what i thought we were asking for. (but the checkbox+warning should only show up when the operator asks for monitoring)

Just to be extra-clear, that is this checkbox and (to be strengthened) note/warning?
![enable monitoring2](https://user-images.githubusercontent.com/54860125/72106748-f19e0080-32fd-11ea-9eb9-6d30f64c0fd0.png)


> > i think the only way to do this is the whitelist you proposed and i'd hate to see us try to maintain that as more and more redhat products start producing operators.
> 
> Yes agreed, as long as there is a warning to the users that is clear enough that installing anything outside of the redhat operators into an openshift- namespace is not a supported feature with the warning and checkbox, we are good to go. 👍

Is the first-mentioned "warning to the users" a different warning? Would that appear any time an operator is installed to openshift-* namespaces? Is the "warning and checkbox" the same as the image in this comment? Sorry, not sure I followed this or @s-urbaniak 's comment.

Reply:
in the current screenshot, this is the warning to users:
`Note: enabling monitoring will allow any operator or workload........`

That's the only warning we are talking about.  (Though as i've said elsewhere, we probably need to reword that warning to make it stronger/clearer what the implications/risks are if you do this)